### PR TITLE
ops: fixing specializing_iet

### PR DESCRIPTION
**Pull request by btrb**
This PR fixes the case in which there isn't affine trees to be converted
into OPS.

There are some fixings on ops/compiler as well.

### CODE REVIEW

File: devito/ops/compiler.py [hunk 1]
```diff
@@ -14,6 +14,9 @@ class CompilerOPS(configuration['compiler'].__class__):
     def __init__(self, *args, **kwargs):
         kwargs['cpp'] = True
         self._ops_install_path = os.environ.get('OPS_INSTALL_PATH')
+        if not self._ops_install_path:
+            raise ValueError("Couldn't find OPS_INSTALL_PATH \
+                environment variable, please check your OPS installation")
```
Comment (maelso) [inline devito/ops/compiler.py]: We are temporarily replacing raise error with warning.

Comment (btrb) [reply devito/ops/compiler.py]: ok ! thanks. fixing

File: devito/ops/operator.py [hunk 2]
```diff
@@ -46,7 +46,7 @@ def _specialize_iet(self, iet, **kwargs):
         # Extract all symbols that need to be converted to ops_dat
         dims = []
         to_dat = set()
-        for section, trees in find_affine_trees(iet).items():
+        for _, trees in affine_trees:
```
Comment (FabioLuporini) [inline devito/ops/operator.py]: nitpicking: I'd use `tree`, not `trees` here

File: devito/ops/operator.py [hunk 3]
```diff
@@ -76,11 +76,13 @@ def _specialize_iet(self, iet, **kwargs):
...
             # from the dat object back to the CPU memory.
-            after_time_loop.extend(create_ops_fetch(f, name_to_ops_dat, time_upper_bound))
+            after_time_loop.extend(create_ops_fetch(f,
+                                                    name_to_ops_dat,
+                                                    self.time_dimmension.extreme_max))
 
         # Generate ops kernels for each offloadable iteration tree
         mapper = {}
-        for n, (section, trees) in enumerate(find_affine_trees(iet).items()):
+        for n, (_, trees) in enumerate(affine_trees):
```
Comment (FabioLuporini) [inline devito/ops/operator.py]: same

File: devito/ops/operator.py [hunk 4]
```diff
@@ -112,3 +114,10 @@ def _compile(self):
         self._includes.append('%s.h' % self._soname)
         if self._lib is None:
             self._compiler.jit_compile(self._soname, str(self.ccode), str(self.hcode))
+
+    @cached_property
+    def time_dimmension(self):
```
Comment (mloubout) [inline devito/ops/operator.py]: typo

Comment (btrb) [reply devito/ops/operator.py]: thank you, fixing

**Comment by FabioLuporini** [comment]
@maelso should I proceed and merge this?

File: devito/ops/compiler.py [hunk 5]
```diff
@@ -47,11 +47,11 @@ def jit_compile(self, soname, ccode, hcode):
             with open(self.ops_src, 'r') as f:
                 self.code = f.read()
         except FileNotFoundError:
-            warning("The file %s isn't present" % self.ops_src)
+            warning("Coudn't find file: %s" % self.ops_src)
```
Comment (maelso) [inline devito/ops/compiler.py]: Typo?

File: devito/ops/compiler.py [hunk 6]
```diff
@@ -80,7 +81,7 @@ def __compile_cuda(self, soname):
             with open(cuda_src, 'r') as f:
                 cuda_code = f.read()
         except FileNotFoundError:
-            raise ValueError("The file %s isn't present" % cuda_src)
+            raise ValueError("Coudn't find file: %s" % cuda_src)
```
Comment (maelso) [inline devito/ops/compiler.py]: Typo again.

File: devito/ops/compiler.py (hunk 5 above)
Comment (btrb) [reply devito/ops/compiler.py]: ty fixing

**Comment by maelso** [comment]
Looks good.

File: devito/ops/operator.py [hunk 7]
```diff
@@ -32,12 +33,11 @@ def __init__(self, *args, **kwargs):
...
         warning("The OPS backend is still work-in-progress")
 
-        # If there is no iteration tree, then there is no loop to be optimized using OPS.
-        iteration_tree = retrieve_iteration_tree(iet, mode='normal')
-        if not len(iteration_tree):
+        affine_trees = find_affine_trees(iet).items()
+
+        # If there is no affine trees, then there is no loop to be optimized using OPS.
+        if not affine_trees:
             return iet
```
Comment (maelso) [inline devito/ops/operator.py]: @btrb The iet will not be specialized for backend OPS, but will still call the CompilerOPS, causing an error. How do you want to proceed?

Comment (FabioLuporini) [reply devito/ops/operator.py]: I'm proceeding with merging, this can be addressed in another PR

**Pull request merged**